src/main/scala/org/hammerlab/guacamole/readsets/rdd/PartitionedRegions.scala

```diff
@@ -1,98 +1,214 @@
- * partition corresponds to.
- *
- * [[WindowFlatMapUtils]] calls this, and uses [[SlidingWindow]] to iterate over the partitioned regions in a manner
- * suitable for pileup-construction by [[PileupFlatMapUtils]].
+ * This means some regions will occur multiple times in the RDD (due to regions straddling partition boundaries, so it's
+ * important not to confuse this with a regular RDD[ReferenceRegion].
  */
+class PartitionedRegions[R <: ReferenceRegion: ClassTag](val numSamples: NumSamples,
+                                                         @transient keyedRegions: RDD[(SampleId, R)],
+                                                         @transient partitioning: LociPartitioning)
+  extends Serializable {
+
+  assert(
+    keyedRegions.getNumPartitions == lociSetsRDD.getNumPartitions,
+    s"reads partitions: ${keyedRegions.getNumPartitions}, loci partitions: ${lociSetsRDD.getNumPartitions}"
+  )
+
+  def sc: SparkContext = keyedRegions.sparkContext
+
+  val numLoci = sc.accumulator(0L, "numLoci")
+
+  def mapPartitions[V: ClassTag](f: (Iterator[(SampleId, R)], LociSet) => Iterator[V]): RDD[V] =
+    keyedRegions
+      .zipPartitions(
+        lociSetsRDD,
+        preservesPartitioning = true
+      )(
+        (regionsIter, lociIter) => {
+          val loci = lociIter.next()
+          if (lociIter.hasNext) {
+            throw new Exception(s"Expected 1 LociSet, found ${1 + lociIter.size}.\n$loci")
+          }
+
+          numLoci += loci.count
+
+          f(regionsIter, loci)
+        }
+      )
+
+  private lazy val partitionLociSets: Array[LociSet] =
+    partitioning
+      .inverse
+      .toArray
+      .sortBy(_._1)
+      .map(_._2)
+
+  lazy val lociSetsRDD: RDD[LociSet] =
+    sc
+      .parallelize(partitionLociSets, partitionLociSets.length)
+      .setName("lociSetsRDD")
+}
+
 object PartitionedRegions {
+
+  type IntHist = MagicHashMap[Int, Long]
+
+  def IntHist(): IntHist = MagicHashMap[Int, Long]()
+
   /**
-   * @param regionRDDs RDDs of regions to partition.
-   * @param lociPartitionsBoxed Spark broadcast of a [[LociPartitioning]] describing which ranges of genomic loci should
-   *                            be mapped to each Spark partition.
+   * @param regionRDDs RDD of regions to partition.
+   * @param loci A [[LociPartitioning]] describing which ranges of genomic loci should be mapped to each Spark
+   *             partition.
+   * @param args Parameters dictating how `loci` should be partitioned.
+   * @tparam R ReferenceRegion type.
+   */
+  def apply[R <: ReferenceRegion: ClassTag](regionRDDs: PerSample[RDD[R]],
+                                            loci: LociSet,
+                                            args: LociPartitionerArgs): PartitionedRegions[R] =
+    apply(regionRDDs, loci, args, halfWindowSize = 0)
+
+  /**
+   * @param regionRDDs RDD of regions to partition.
+   * @param loci A [[LociPartitioning]] describing which ranges of genomic loci should be mapped to each Spark
```

Comment:
In the code lLoci` seems to be a `LociSet` not a `LociPartitioning`. (I also find that confusing though -- doesn't this want some kind of map from LociSets to partitions?) A one sentence intro comment to this function would be helpful

Also same for the override above.